src/Hosting/Hosting/src/WebHostBuilderExtensions.cs

```diff
@@ -151,7 +155,7 @@ public static IWebHostBuilder UseStartup(this IWebHostBuilder hostBuilder, Type
         /// <param name="hostBuilder">The <see cref="IWebHostBuilder"/> to configure.</param>
         /// <typeparam name ="TStartup">The type containing the startup methods for the application.</typeparam>
         /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
-        public static IWebHostBuilder UseStartup<TStartup>(this IWebHostBuilder hostBuilder) where TStartup : class
+        public static IWebHostBuilder UseStartup<[DynamicallyAccessedMembers(StartupAccessibility)]TStartup>(this IWebHostBuilder hostBuilder) where TStartup : class
```

Comment:
nit: is there any guidance or best practice around generic attributes and overloads. It looks like this method calls the earlier UseStartup which will have the attribute so it probably isn't necessary. Not a big deal either way.

Reply:
I think I'm supposed to apply this attribute transitively through the codebase or the linker will warn.